Riot/Modules/Home/HomeViewController.m

```diff
@@ -114,7 +114,7 @@ - (void)viewWillAppear:(BOOL)animated
         [recentsDataSource setDelegate:self andRecentsDataSourceMode:RecentsDataSourceModeHome];
         
         // Reset filtering on the shared data source when switching tabs
-        [self searchBarCancelButtonClicked:self.recentsSearchBar];
+        [recentsDataSource searchWithPatterns:nil];
```

Comment:
This basically makes it the same as all the other tabs in which it worked properly.